### ISSUE TITLE
How could I get the loss of test datasets ?

### ISSUE THREAD

**Issue by wkoa** [facebookresearch/Detectron#315]
I have checked the code, but I can not find a function, whose input is model and datasets with its output is the loss of datasets in which I use the model of inputs to run testing task.

**Comment by ir413** [comment]
Hi @wkoa, we don't construct the losses at test-time (e.g. see [this](https://github.com/facebookresearch/Detectron/blob/master/lib/modeling/model_builder.py#L257-L260) for cls and bbox head losses). If you would like to compute the losses for test images you will need to construct the losses at test-time. Note that during training we combine the softmax operator with loss for numerical stability (see [this](https://github.com/facebookresearch/Detectron/blob/master/lib/modeling/fast_rcnn_heads.py#L55-L58)).
